epochStart/shardchain/triggerRegistry_test.go

```diff
@@ -135,30 +136,167 @@ func TestTrigger_LoadStateBackwardsCompatibility(t *testing.T) {
 	t.Parallel()
 
 	epoch := uint32(5)
-	arguments := createMockShardEpochStartTriggerArguments()
-	arguments.Epoch = epoch
-	bootStorer := genericMocks.NewStorerMock()
+	key := []byte("key")
 
-	arguments.Storage = &storageStubs.ChainStorerStub{
-		GetStorerCalled: func(unitType dataRetriever.UnitType) (storage.Storer, error) {
-			return bootStorer, nil
-		},
-	}
+	t.Run("backwards compatibility", func(t *testing.T) {
+		t.Parallel()
 
-	key := []byte("key")
-	epochStartTrigger1, epochStartTrigger2 := createDummyEpochStartTriggers(arguments, key)
+		arguments := createMockShardEpochStartTriggerArguments()
+		arguments.Epoch = epoch
+		bootStorer := genericMocks.NewStorerMock()
 
-	trig := createLegacyTriggerRegistryFromTrigger(epochStartTrigger1)
-	d, err := json.Marshal(trig)
-	require.Nil(t, err)
-	trigInternalKey := append([]byte(common.TriggerRegistryKeyPrefix), key...)
+		arguments.Storage = &storageStubs.ChainStorerStub{
+			GetStorerCalled: func(unitType dataRetriever.UnitType) (storage.Storer, error) {
+				return bootStorer, nil
+			},
+		}
 
-	err = bootStorer.Put(trigInternalKey, d)
-	require.Nil(t, err)
+		epochStartTrigger1, epochStartTrigger2 := createDummyEpochStartTriggers(arguments, key)
 
-	err = epochStartTrigger2.LoadState(key)
-	require.Nil(t, err)
-	require.Equal(t, epochStartTrigger1, epochStartTrigger2)
+		trig := createLegacyTriggerRegistryFromTrigger(epochStartTrigger1)
+		d, err := json.Marshal(trig)
+		require.Nil(t, err)
+		trigInternalKey := append([]byte(common.TriggerRegistryKeyPrefix), key...)
+
+		err = bootStorer.Put(trigInternalKey, d)
+		require.Nil(t, err)
+
+		err = epochStartTrigger2.LoadState(key)
+		require.Nil(t, err)
+		require.Equal(t, epochStartTrigger1, epochStartTrigger2)
+	})
+
+	t.Run("header v1", func(t *testing.T) {
```

Comment:
I'm personally not a big fan of this much duplicated code.
You could've made a base test func with interfaces for Header + trigger and pass actual pointers to v1,v2,v3